pkg/providers/instance/instance.go

```diff
@@ -670,6 +705,54 @@ func (p *Provider) getAKSIdentifyingExtension() *armcompute.VirtualMachineExtens
 	return vmExtension
 }
 
+func (p *Provider) getCSExtension(cse string, isWindows bool) *armcompute.VirtualMachineExtension {
+	const (
+		vmExtensionType     = "Microsoft.Compute/virtualMachines/extensions"
+		cseNameWindows      = "windows-cse-agent-karpenter"
+		cseTypeWindows      = "CustomScriptExtension"
+		csePublisherWindows = "Microsoft.Compute"
+		cseVersionWindows   = "1.10"
+		cseNameLinux        = "cse-agent-karpenter"
+		cseTypeLinux        = "CustomScript"
+		csePublisherLinux   = "Microsoft.Azure.Extensions"
+		cseVersionLinux     = "2.0"
```

Comment:
How often do these values change? what exactly is cseVersionLinux?

Reply:
Barely changes.